### PR TITLE
[Feature][BugFix][Spec Decode] Support DSpark with LMHead TP

### PR DESCRIPTION
**Pull request by jiajinzhu2**
### What this PR does / why we need it?

This PR fixes DSpark speculative decoding when fine-grained LMHead tensor parallelism is enabled.

- Keep the DSpark Markov embedding and low-rank vocabulary projection replicated. Markov decoding is serial across draft positions, so sharding these weights through the LMHead TP group adds an unintended collective to every Markov step.
- Pad the DSpark LMHead input with the draft width `K`, instead of the target runner decode-query width `K + 1`.
- Remove flat dummy sampling rows immediately after the LMHead TP collective and before reshaping logits to `[batch, K, vocab]`.

Without these changes, DSpark uses the generic LMHead TP path with incompatible draft-side shapes and dummy request blocks can reach Markov decoding.

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

```
No LMHead TP                                          │ LMHead TP=4
======================================================│======================================================
============ Serving Benchmark Result ===============│============ Serving Benchmark Result ===============
Successful requests:                     100          │Successful requests:                     100
Failed requests:                         0            │Failed requests:                         0
Maximum request concurrency:             4            │Maximum request concurrency:             4
Benchmark duration (s):                  136.02       │Benchmark duration (s):                  127.96
Total input tokens:                      26148        │Total input tokens:                      26148
Total generated tokens:                  31188        │Total generated tokens:                  31394
Request throughput (req/s):              0.74         │Request throughput (req/s):              0.78
Output token throughput (tok/s):         229.29       │Output token throughput (tok/s):         245.34
Peak output token throughput (tok/s):    84.00        │Peak output token throughput (tok/s):    80.00
Peak concurrent requests:                6.00         │Peak concurrent requests:                7.00
Total token throughput (tok/s):          421.52       │Total token throughput (tok/s):          449.68
---------------Time to First Token-------------------│---------------Time to First Token-------------------
Mean TTFT (ms):                          474.37       │Mean TTFT (ms):                          461.21
Median TTFT (ms):                        432.57       │Median TTFT (ms):                        422.38
P50 TTFT (ms):                           432.57       │P50 TTFT (ms):                           422.38
P75 TTFT (ms):                           466.12       │P75 TTFT (ms):                           455.02
P90 TTFT (ms):                           696.29       │P90 TTFT (ms):                           653.06
P99 TTFT (ms):                           841.05       │P99 TTFT (ms):                           888.03
-----Time per Output Token (excl. 1st token)---------│-----Time per Output Token (excl. 1st token)---------
Mean TPOT (ms):                          15.55        │Mean TPOT (ms):                          14.34
Median TPOT (ms):                        15.55        │Median TPOT (ms):                        14.60
P50 TPOT (ms):                           15.55        │P50 TPOT (ms):                           14.60
P75 TPOT (ms):                           16.73        │P75 TPOT (ms):                           15.34
P90 TPOT (ms):                           17.94        │P90 TPOT (ms):                           16.24
P99 TPOT (ms):                           20.13        │P99 TPOT (ms):                           17.65
---------------Inter-token Latency-------------------│---------------Inter-token Latency-------------------
Mean ITL (ms):                           64.76        │Mean ITL (ms):                           59.78
Median ITL (ms):                         55.38        │Median ITL (ms):                         50.40
P50 ITL (ms):                            55.38        │P50 ITL (ms):                            50.40
P75 ITL (ms):                            56.58        │P75 ITL (ms):                            51.58
P90 ITL (ms):                            58.30        │P90 ITL (ms):                            53.97
P99 ITL (ms):                            371.27       │P99 ITL (ms):                            336.08
----------------End-to-end Latency-------------------│----------------End-to-end Latency-------------------
Mean E2EL (ms):                          5363.61      │Mean E2EL (ms):                          5014.37
Median E2EL (ms):                        4943.06      │Median E2EL (ms):                        4471.54
P50 E2EL (ms):                           4943.06      │P50 E2EL (ms):                           4471.54
P75 E2EL (ms):                           7161.59      │P75 E2EL (ms):                           6990.36
P90 E2EL (ms):                           8997.23      │P90 E2EL (ms):                           8441.55
P99 E2EL (ms):                           10116.44     │P99 E2EL (ms):                           9485.92
---------------Speculative Decoding------------------│---------------Speculative Decoding------------------
Acceptance rate (%):                     63.79        │Acceptance rate (%):                     63.55
Acceptance length:                       4.19         │Acceptance length:                       4.18
Drafts:                                  7450         │Drafts:                                  7517
Draft tokens:                            37250        │Draft tokens:                            37585
Accepted tokens:                         23762        │Accepted tokens:                         23885
Per-position acceptance (%):                          │Per-position acceptance (%):
  Position 0:                            88.52        │  Position 0:                            88.45
  Position 1:                            75.10        │  Position 1:                            74.98
  Position 2:                            62.75        │  Position 2:                            62.46
  Position 3:                            51.65        │  Position 3:                            51.32
  Position 4:                            40.93        │  Position 4:                            40.53
======================================================│======================================================
```

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

**Comment by drslark** [comment]
Add [Feature] to title.

**Review by drslark — APPROVED** [review]
LGTM.

**Review by ningjingbengxiaohai — APPROVED** [review]
